File: voice-vault-frontend/types/audio.ts
export type AudioStatus = "PROCESSING" | "READY" | "FAILED";

export interface AudioTrack {
  id: string;
  noteId: string;
  title: string;
  status: AudioStatus;
  errorMessage: string | null;
  durationSeconds: number | null;
  category: string;
  sourceFileName: string;
  createdAt: string;
}

export interface AudioDownload {
  url: string;
}

File: voice-vault-backend/app/schemas/audio.py
import uuid
from datetime import datetime

from app.schemas.base import CamelModel


class AudioOut(CamelModel):
    id: uuid.UUID
    note_id: uuid.UUID
    title: str
    status: str
    error_message: str | None
    duration_seconds: int | None
    category: str
    source_file_name: str
    created_at: datetime


class AudioDownloadOut(CamelModel):
    url: str
